resource/composed/composed.go

```diff
@@ -42,6 +42,34 @@ func New() *Unstructured {
 	return &Unstructured{unstructured.Unstructured{Object: make(map[string]any)}}
 }
 
+// To converts a unstructured composed resource to the provided object.
+func To[T runtime.Object](un *Unstructured, obj T) error {
+
+	// Get known GVKs for the runtime object type
+	knownGVKs, _, err := Scheme.ObjectKinds(obj)
+	if err != nil {
+		return errors.Errorf("could not retrieve GVKs for the provided object: %v", err)
+	}
+
+	// Check if GVK is known as we should not try to convert it if it doesn't match
+	gvkMatches := false
+	for _, knownGVK := range knownGVKs {
+		if knownGVK == un.GetObjectKind().GroupVersionKind() {
+			gvkMatches = true
+		}
+	}
+
+	if !gvkMatches {
+		return errors.Errorf("GVK %v is not known by the scheme for the provided object type", un.GetObjectKind().GroupVersionKind())
+	}
+
+	err = runtime.DefaultUnstructuredConverter.FromUnstructured(un.Object, obj)
+	if err != nil {
+		return err
+	}
+	return nil
```

Comment:
```suggestion
	return runtime.DefaultUnstructuredConverter.FromUnstructured(un.Object, obj)
```

This is equivalent to what you have.

It'd be ideal to wrap the error here too. `return errors.Wrap(runtime.Default..., "...")` will return `nil` if the wrapped call returns a `nil` error.